### PR TITLE
Closes #267 Add nav-pills flex-column styles

### LINKED ISSUE #267: Update sidebar nav styles
Update sidebar nav styles on Arizona Bootstrap
```
ul.nav.nav-pills.flex-column>li {
    border: solid 1px $bg-gray-200;
    margin: -1px 0 0 0;
}
ul.nav.nav-pills.flex-column>li>a:hover,
ul.nav-pills.flex-column>li>a:focus {
    background-color: $bg-gray-200;
}
ul.nav.nav-pills.flex-column>li>ul>li {
    border: none;
}
ul.nav.nav-pills.flex-column>li>ul>li>a {
    font-weight: 400;
    color: $blue;
    padding-left: 25px;
}
```
$bg-gray-200 = #e9ecef
$blue = #0c234b

### PR DESCRIPTION
**Pull request by danahertzberg**
Closes #267 

Updates sidebar nav styles to match QS1

Review site: https://review.digital.arizona.edu/arizona-bootstrap/issue/267/docs/2.0/components/navs/#vertical-pills

### CODE REVIEW

**Comment by danahertzberg** [comment]
I'm going to have to ask for help with this one..
```
scss/_nav.scss
 46:5  ✖  Expected ".nav.nav-pills.flex-column > li ul li" to have no more than 2 type selectors         selector-max-type              
 47:5  ✖  Expected indentation of 6 spaces                                                               indentation                    
 48:7  ✖  Expected ".nav.nav-pills.flex-column > li ul li a" to have no more than 4 compound selectors   selector-max-compound-selectors
 48:7  ✖  Expected ".nav.nav-pills.flex-column > li ul li a" to have no more than 2 type selectors       selector-max-type

File: scss/_nav.scss [hunk 1]
```diff
@@ -35,6 +35,25 @@
...
+.nav.nav-pills.flex-column > li {
+  margin: -1px 0 0;
+  border: solid 1px $nav-pills-link-active-bg;
+  a:hover,
+  a:focus {
+    background-color: $nav-pills-link-active-bg;
+  }
+  ul {
+    li {
+    border: none;
```
Comment (joeparsons) [inline scss/_nav.scss:47]:
```suggestion
      border: none;
```

File: scss/_nav.scss [hunk 2]
```diff
@@ -35,6 +35,25 @@
   }
 }
 
+.nav.nav-pills.flex-column > li {
```
Comment (joeparsons) [inline scss/_nav.scss:38]: Maybe doing the selector inheritance with sass would help with the linting errors here (?).

i.e. something like this:
```
.nav {
  .nav-pills {
    .flex-column > li {
```

**Comment by mmunro-ltrr** [comment]
The current issue/267 branch is based on quite an old commit: rebasing locally to look at various things, but I might make an alternative version.

**Comment by joeparsons** [comment]
> I'm going to have to ask for help with this one..
> 
> ```
> scss/_nav.scss
>  46:5  ✖  Expected ".nav.nav-pills.flex-column > li ul li" to have no more than 2 type selectors         selector-max-type              
>  47:5  ✖  Expected indentation of 6 spaces                                                               indentation                    
>  48:7  ✖  Expected ".nav.nav-pills.flex-column > li ul li a" to have no more than 4 compound selectors   selector-max-compound-selectors
>  48:7  ✖  Expected ".nav.nav-pills.flex-column > li ul li a" to have no more than 2 type selectors       selector-max-type
> ```

https://stylelint.io/user-guide/rules/selector-max-type
https://stylelint.io/user-guide/rules/selector-max-compound-selectors

**Comment by joeparsons** [comment]
https://stylelint.io/user-guide/ignore-code

File: scss/_nav.scss [hunk 3]
```diff
@@ -35,6 +35,27 @@
   }
 }
 
+.nav.nav-pills.flex-column {
```
Comment (joeparsons) [inline scss/_nav.scss:38]:
```suggestion
/* stylelint-disable selector-max-type, selector-max-compound-selectors */
.nav.nav-pills.flex-column {
```

File: scss/_nav.scss [hunk 4]
```diff
@@ -35,6 +35,27 @@
...
+        border: none;
+        a {
+          padding-left: 25px;
+          font-weight: 400;
+          color: $nav-pills-link-active-color;
+        }
+      }
+    }
+  }
+}
```
Comment (joeparsons) [inline scss/_nav.scss:57]:
```suggestion
}
/* stylelint-enable selector-max-type, selector-max-compound-selectors */
```

**Comment by mmunro-ltrr** [comment]
That's encouraging! Re-starting the local review copy.

**Comment by danahertzberg** [comment]
> That's encouraging! Re-starting the local review copy.

Woo hoo! I realized I still need to add an example to the docs to actually see my changes. Working on that now. :)

File: scss/_nav.scss [hunk 5]
```diff
@@ -35,6 +35,57 @@
...
+        border: none;
+        a {
+          padding-left: 25px;
+          font-weight: 400;
+          color: $nav-pills-link-active-color;
+        }
+        ul {
+          li {
+            a {
+            padding-left: 35px;
```
Comment (joeparsons) [inline scss/_nav.scss:58]:
```suggestion
              padding-left: 35px;
```

File: scss/_nav.scss [hunk 6]
```diff
@@ -35,6 +35,57 @@
...
+            }
+            ul {
+              li {
+                a {
+                padding-left: 45px;
```
Comment (joeparsons) [inline scss/_nav.scss:63]:
```suggestion
                  padding-left: 45px;
```

File: scss/_nav.scss [hunk 7]
```diff
@@ -35,6 +35,57 @@
...
+                }
+                ul {
+                  li {
+                    a {
+                    padding-left: 55px;
```
Comment (joeparsons) [inline scss/_nav.scss:68]:
```suggestion
                      padding-left: 55px;
```

File: scss/_nav.scss [hunk 8]
```diff
@@ -35,6 +35,57 @@
...
+                    }
+                    ul {
+                      li {
+                        a {
+                        padding-left: 65px;
```
Comment (joeparsons) [inline scss/_nav.scss:73]:
```suggestion
                          padding-left: 65px;
```

File: scss/_nav.scss [hunk 9]
```diff
@@ -35,6 +35,57 @@
   }
 }
 
+/* stylelint-disable selector-max-type, selector-max-compound-selectors */
```
Comment (joeparsons) [inline scss/_nav.scss:38]:
```suggestion
/* stylelint-disable selector-max-type, selector-max-compound-selectors, selector-max-combinators */
```

File: scss/_nav.scss [hunk 10]
```diff
@@ -35,6 +35,57 @@
...
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+/* stylelint-enable selector-max-type, selector-max-compound-selectors */
```
Comment (joeparsons) [inline scss/_nav.scss:87]:
```suggestion
/* stylelint-enable selector-max-type, selector-max-compound-selectors, selector-max-combinators */
```

**Comment by danahertzberg** [comment]
This is finally ready for review! Review site updated in the description.

**Comment by joeparsons** [comment]
The padding needs to be on the `<a>` elements (not the `<li>` elements), right?  I was just thinking if the padding could be applied to the `<li>` element it would reduce the amount of selectors needed...

**Comment by danahertzberg** [comment]
> The padding needs to be on the `<a>` elements (not the `<li>` elements), right? I was just thinking if the padding could be applied to the `<li>` element it would reduce the amount of selectors needed...

@joeparsons Good question. I just checked, and yes it has to be on the `<a>` element in order for the background color to span the entire width on hover and active states.

**Comment by danahertzberg** [comment]
@joeparsons I could use a `ul > li > ul > li > a {` selector to reduce the number of lines and curly braces. What do you think?

**Comment by joeparsons** [comment]
> @joeparsons I could use a `ul > li > ul > li > a {` selector to reduce the number of lines and curly braces. What do you think?

Sounds reasonable.  Probably won't fix all the stylelint errors but it might fix some of them (you'd have to try removing the rules from the `stylelint-disable` line to check.

**Review by smayden — APPROVED** [review]
This looks good to me from a front-end perspective on the review site.

**Review by mmunro-ltrr — APPROVED** [review]
Checked on a local build.

**Review by tadean — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Pull request merged**